odl/solvers/functional/default_functionals.py

```diff
@@ -2470,6 +2470,149 @@ def __repr__(self):
+        --------
+        Example of initializing the Bregman distance functional:
+
+        >>> space = odl.uniform_discr(0, 2, 14)
+        >>> l2_squared = odl.solvers.L2NormSquared(space)
+        >>> point = space.one()
+        >>> Bregman_dist = odl.solvers.BregmanDistance(l2_squared, point)
+
+        This is gives the shifted L2 norm squared ||x - 1||:
+
+        >>> Bregman_dist(space.zero())
+        2.0
+        """
+        if not isinstance(functional, Functional):
+            raise TypeError('`functional` {} not an instance of ``Functional``'
+                            ''.format(functional))
+        self.__functional = functional
+
+        if point not in functional.domain:
+            raise ValueError('`point` {} is not in `functional.domain` {}'
+                             ''.format(point, functional.domain))
+        self.__point = point
+
+        if subgradient_op is None:
+            # If the subgradient is not given, assign the gradient of the
+            # functional
+            try:
+                self.__subgradient_op = self.__functional.gradient
+            except NotImplementedError:
+                raise NotImplementedError('`subgradient` not given, and the '
+                                          '`functional` {} does not implement '
+                                          '`functional.gradient`'
+                                          ''.format(functional))
+        else:
+            # Check that given subgradient is an operator that maps from the
+            # domain of the functional to the domain of the functional
+            if not isinstance(subgradient_op, Operator):
+                raise TypeError('`subgradient_op` {} is not an instance of '
+                                '``Operator``'.format(subgradient_op))
+            if not self.__functional.domain == subgradient_op.domain:
+                raise ValueError('`functional.domain` {} is not the same as '
+                                 '`subgradient_op.domain` {}'
+                                 ''.format(self.__functional.domain,
+                                           subgradient_op.domain))
+            if not self.__functional.domain == subgradient_op.range:
+                raise ValueError('`functional.domain` {} is not the same as '
+                                 '`subgradient_op.range` {}'
+                                 ''.format(self.__functional.domain,
+                                           subgradient_op.range))
+            self.__subgradient_op = subgradient_op
+
+        self.__subgrad_eval = self.__subgradient_op(self.__point)
+        self.__bregman_dist = (FunctionalQuadraticPerturb(
+            self.__functional, linear_term=-self.__subgrad_eval) -
+            self.__functional(point) + self.__subgrad_eval.inner(point))
+
+        super(BregmanDistance, self).__init__(
+            space=functional.domain, linear=False,
+            grad_lipschitz=self.__functional.grad_lipschitz)
```

Comment:
Is this really correct? Shouldn't the linear term add something

Reply:
You are absolutely right! But `grad D_f(x, y) = grad f(x) - grad f(y)`, so it is to me not obvious how much it adds... however by the triangle inequality it should be bounded by the `grad_lipschitz` + the norm of `grad f(y)`, right? Maybe that is ok as an estimate?